### PR TITLE
Support translation of JsonPathType to ConnectorExpression

### PR DESCRIPTION
**Pull request by assaf2**
## Description
Support translation of `JsonPathType` (which is not part of SPI) to `ConnectorExpression` in order to support pushdown of functions like `json_extract_scalar`.

## Release notes

( X ) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

### CODE REVIEW

**Comment by findepi** [comment]
The `TestPredicatePushdown>AbstractPredicatePushdownTest.testDoesNotCreatePredicateFromInferredPredicate:` failure looks related.

**Comment by assaf2** [comment]
> The `TestPredicatePushdown>AbstractPredicatePushdownTest.testDoesNotCreatePredicateFromInferredPredicate:` failure looks related.

Rebase fixed the issue

File: core/trino-main/src/test/java/io/trino/sql/planner/TestConnectorExpressionTranslator.java [hunk 1]
```diff
@@ -455,6 +456,39 @@ public void testTranslateRegularExpression()
...
+
+        TransactionManager transactionManager = new TestingTransactionManager();
+        Metadata metadata = MetadataManager.testMetadataManagerBuilder().withTransactionManager(transactionManager).build();
+        transaction(transactionManager, metadata, new AllowAllAccessControl())
+                .readOnly()
+                .execute(TEST_SESSION, transactionSession -> {
+                    FunctionCall input = BuiltinFunctionCallBuilder.resolve(PLANNER_CONTEXT.getMetadata())
+                            .setName("json_extract_scalar")
+                            .addArgument(VARCHAR_TYPE, new SymbolReference("varchar_symbol_1"))
+                            .addArgument(JSON_PATH, new StringLiteral("$.path"))
```
Comment (findepi) [inline core/trino-main/src/test/java/io/trino/sql/planner/TestConnectorExpressionTranslator.java:473]: should this be cast to JsonPathType?

File: core/trino-main/src/test/java/io/trino/sql/planner/TestConnectorExpressionTranslator.java [hunk 2]
```diff
@@ -455,6 +456,39 @@ public void testTranslateRegularExpression()
                 });
     }
 
+    @Test
+    void testTranslateJsonPath()
+    {
+        // JsonPath type is considered implementation detail of the engine
```
Comment (findepi) [inline core/trino-main/src/test/java/io/trino/sql/planner/TestConnectorExpressionTranslator.java:462]: `JsonPath` looks like a formal name, but the name is actually `JsonPathType`

Let's maybe write it as "JSON path type"

**Review by findepi — APPROVED** [review]

**Review by findepi — APPROVED** [review]

**Pull request merged**